### ISSUE TITLE
Wrap on multi-caching doesn't work when no data is returned

### ISSUE THREAD

**Issue by rubenstolk** [BryanDonovan/node-cache-manager#89]
See the short snippet below. The last `console.log` is never being executed because the `getEmpty` function doesn't return any data.

This is a major issue as it can result in a complete `hang` of an application.

```js
const manager = require('cache-manager');

const cache = manager.multiCaching([
  manager.caching({ ttl: 300, store: 'memory' }),
  manager.caching({ ttl: 300, store: 'memory' })
]);

const key1 = 'key1';
const key2 = 'key2';

const getData = fn => setTimeout(() => fn(null, 'SOME_DATA'), 1000);
const getEmpty = fn => setTimeout(() => fn(null), 1000);

cache.wrap(key2, getData, (err, data) =>
  console.log(`RETURN FROM WRAP WITH DATA ${key1} (1)`, err, data));

cache.wrap(key2, getData, (err, data) =>
  console.log(`RETURN FROM WRAP WITH DATA ${key1} (2)`, err, data));

cache.wrap(key1, getEmpty, (err, data) =>
  console.log(`RETURN FROM WRAP WITHOUT DATA ${key2} (1)`, err, data));

cache.wrap(key1, getEmpty, (err, data) =>
  console.log(`RETURN FROM WRAP WITHOUT DATA ${key2} (2)`, err, data));
```

**Comment by rubenstolk** [comment]
FYI this is the work around that I've had to apply:

```js
// Workaround for https://github.com/BryanDonovan/node-cache-manager/issues/89
cache._wrap = module.exports.wrap;
cache.wrap = function (key, work, options, cb) {
  const _work = fn => work((err, data) => fn(err, data || null));
  return cache._wrap.call(this, key, _work, options, cb);
};
```

**Comment by BryanDonovan** [comment]
I believe overriding the `isCacheableValue` would work:

```javascript
const cache = manager.multiCaching([
    manager.caching({ttl: 300, store: 'memory'}),
    manager.caching({ttl: 300, store: 'memory'})],
    {
        isCacheableValue: value => {
            return true;
        }
    }
);
```

.. but that would also persist (or try to persist) `undefined` in the cache store.

Maybe the real solution is to have a `isReturnableValue` function as mentioned here: https://github.com/BryanDonovan/node-cache-manager/pull/42

**Comment by rubenstolk** [comment]
Don't you think it's weird that the functionality is different between using `.wrap` on `multi-caching` and `wrap` on an individual store?

**Comment by BryanDonovan** [comment]
Ah, yes that is weird.  I'm looking at the code and not seeing why it behaves differently between the multi and individual stores.  Do you see what's wrong there?

The `null` vs `undefined` thing is strange (and older versions of this library treated them equally), but `null` and `undefined` mean two entirely different things --  `undefined` is a lack of value whereas `null` can be and often is a value someone set on purpose.  But this is why you can override the `isCacheableValue` function.

**Comment by BryanDonovan** [comment]
This should be fixed now in 2.6.0
